2-projects/1-todo-list/script.js

```diff
@@ -12,14 +30,58 @@ let todos = [
 
 populateTodoList(todos);
 
+function createDOMListItem(todo) {
+  const li = document.createElement("li");
+
+  const tickIcon = document.createElement("i");
+  tickIcon.classList.add("fa", "fa-check");
+  const binIcon = document.createElement("i");
+  binIcon.classList.add("fa", "fa-trash");
+
+  if (!tickIcon.onclick) {
+    tickIcon.addEventListener("click", () => {
+      toggleIsComplete(todo);
+      redrawTodo(li, todo.completed);
+    });
+  }
+  if (!binIcon.onclick) {
+    binIcon.addEventListener("click", () => deleteTodo(li));
+  }
+  if (todo.completed) {
+    li.style.textDecoration = "line-through";
+  }
+
+  li.innerText = todo.task;
+  li.appendChild(tickIcon);
+  li.appendChild(binIcon);
+  return li;
+}
+
+function appendTodo(todo) {
+  let list = document.getElementById("todo-list");
+  const li = createDOMListItem(todo);
+  list.appendChild(li);
+}
+
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+
+  const { value: task } = document.querySelector("#todoInput");
+  appendTodo({ task, completed: false });
 }
 
+
+
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
-function deleteAllCompletedTodos() {
-  // Write your code here...
+function deleteAllCompletedTodos(e) {
+  e.preventDefault();
+  const list = document.querySelector("ul");
+  const completedTodos = [...document.querySelectorAll("li")].filter((todo) => {
+    return todo.style.textDecoration === "line-through";
+  });
+
+  completedTodos.forEach((todo) => {
+    list.removeChild(todo);
+  });
 }
```

Comment:
You should register `deleteAllCompletedTodos` as onclick handler to the remove button:

```
let removeButton = document.getElementById("remove-all-completed");
removeButton.addEventListener("click", deleteAllCompletedTodos);
```